### PR TITLE
Fix errors for VariadicSplit layer restored from serialized IR

### PR DESCRIPTION
**Pull request by achetver**
### Tickets:
 - 55256

Root cause analysis: Since POT starts using serialized IR `axis` check in VariadicSplit shape inference function haven't passed for layers restored from such IR's. It happens because `axis` value in restored IR stored as 1D tensor with 1 element, not as scalar.

Solution: Update guilty check to support this case.

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: Added;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done manually.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Review by yekruglov — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

File: model-optimizer/extensions/ops/split.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def infer(node):
 
         axis = node.in_port(1).data.get_value() if op == 'VariadicSplit' else node.soft_get('axis', None)
         assert axis is not None, '{} `axis` is unknown for node {}'.format(op, name)
-        assert axis.ndim == 0, '{} `axis` should be scalar, but it`s not for node {}'.format(op, name)
+        assert axis.ndim == 0 or len(axis) == 1 and axis.ndim == 1, '{} `axis` should be scalar, but it`s not for node {}'.format(op, name)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/split.py:47]: Update the comment. Now the input could be a scalar or a tensor with shape `[1]`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by evolosen — APPROVED** [review]

File: docs/ops/movement/VariadicSplit_1.md [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 **Detailed Description**
 
-*VariadicSplit* operation splits a given input tensor `data` into chunks along a scalar `axis`. It produces multiple output tensors based on additional input tensor `split_lengths`.
+*VariadicSplit* operation splits a given input tensor `data` into chunks along a scalar (or tensor with shape `[1]`) `axis`. It produces multiple output tensors based on additional input tensor `split_lengths`.
```
Comment (tsavina) [inline docs/ops/movement/VariadicSplit_1.md:11]: along a scalar (or tensor with shape `[1]`) `axis` ---> along a scalar `axis` or tensor with shape `[1]`

File: model-optimizer/extensions/ops/split.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/extensions/ops/split.py]: Done.

File: docs/ops/movement/VariadicSplit_1.md (hunk 2 above)
Comment (achetver) [reply docs/ops/movement/VariadicSplit_1.md]: Done.

**Review by tsavina — APPROVED** [review]

File: model-optimizer/extensions/ops/split.py [hunk 3]
```diff
@@ -44,7 +44,8 @@ def infer(node):
 
         axis = node.in_port(1).data.get_value() if op == 'VariadicSplit' else node.soft_get('axis', None)
         assert axis is not None, '{} `axis` is unknown for node {}'.format(op, name)
-        assert axis.ndim == 0, '{} `axis` should be scalar, but it`s not for node {}'.format(op, name)
+        assert axis.ndim == 0 or len(axis) == 1 and axis.ndim == 1, '{} `axis` should be scalar or tensor with shape ' \
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/split.py:47]: The condition is wrong. The `ndim` attribute contains rank of the tensor. Also, we should wrap the condition with parentheses to explicitly explain the logic. So I suggest to change the code the following way:

```python
assert axis.ndim == 0 or (axis.ndim == 1 and axis[0] == 1), ...
```

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/ops/split.py [hunk 4]
```diff
@@ -44,7 +44,8 @@ def infer(node):
...
+                                                                    '[1], but it`s not for node {}'.format(op, name)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/split.py:48]: Add unit test to test your changes. I have unchecked "unit tests" item in the PR description.

File: model-optimizer/extensions/ops/split.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/extensions/ops/split.py:47]: But your condition is wrong too. According to `VariadicSplit` specification `axis` value (`axis[0]` in check) value can be from range `-rank(data) .. rank(data)-1`. So with such condition we breaks infer function.

Comment (lazarevevgeny) [reply model-optimizer/extensions/ops/split.py:47]: Yes, my bad. The condition should be:

```python
assert axis.ndim == 0 or (axis.ndim == 1 and axis.shape[0] == 1), ...
```

Comment (achetver) [reply model-optimizer/extensions/ops/split.py]: Condition updated.

File: model-optimizer/extensions/ops/split.py (hunk 4 above)
Comment (achetver) [reply model-optimizer/extensions/ops/split.py:48]: Unit tests were added.

File: model-optimizer/unit_tests/extensions/ops/split_test.py [hunk 5]
```diff
@@ -188,3 +188,76 @@ def test_splitv_2_zero_not_last(self):
         self.assertTrue(node.out_port(4).disconnected())
         self.assertTrue(node.out_port(3).disconnected())
         self.assertTrue(np.all(node.split_lengths == np.array([2, 13, 10])))
+
+class TestVariadicSplitOp(unittest.TestCase):
```
Comment (lazarevevgeny) [inline model-optimizer/unit_tests/extensions/ops/split_test.py:192]: Tests are identical except the input shape of the axis. Please, generalize tests using with the generator.
Also, add a test which fails the shape inference, for example, shape of axis is `[2]` or `[1,1]`

File: model-optimizer/extensions/ops/split.py [hunk 6]
```diff
@@ -44,7 +44,8 @@ def infer(node):
 
         axis = node.in_port(1).data.get_value() if op == 'VariadicSplit' else node.soft_get('axis', None)
         assert axis is not None, '{} `axis` is unknown for node {}'.format(op, name)
-        assert axis.ndim == 0, '{} `axis` should be scalar, but it`s not for node {}'.format(op, name)
+        assert axis.ndim == 0 or (axis.ndim == 1 and axis.shape[0] == 1), '{} `axis` should be scalar or tensor with shape ' \
+                                                                    '[1], but it`s not for node {}'.format(op, name)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/split.py:48]: Fix indentation.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/unit_tests/extensions/ops/split_test.py [hunk 7]
```diff
@@ -188,3 +188,76 @@ def test_splitv_2_zero_not_last(self):
         self.assertTrue(node.out_port(4).disconnected())
         self.assertTrue(node.out_port(3).disconnected())
         self.assertTrue(np.all(node.split_lengths == np.array([2, 13, 10])))
+
```
Comment (lazarevevgeny) [inline model-optimizer/unit_tests/extensions/ops/split_test.py:191]: Add one more empty line.

In general, apply pep8 to the whole file.

File: model-optimizer/unit_tests/extensions/ops/split_test.py (hunk 5 above)
Comment (achetver) [reply model-optimizer/unit_tests/extensions/ops/split_test.py:196]: Done, unit tests updated.

File: model-optimizer/extensions/ops/split.py (hunk 6 above)
Comment (achetver) [reply model-optimizer/extensions/ops/split.py]: Done.

File: model-optimizer/unit_tests/extensions/ops/split_test.py (hunk 7 above)
Comment (achetver) [reply model-optimizer/unit_tests/extensions/ops/split_test.py:193]: Done.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**